src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -134,13 +127,33 @@
     public abstract Builder<MethodId> setRetryParams(
         ImmutableMap<MethodId, RetryParams> retryParams);
 
-    public abstract Builder<MethodId> setCredentials(Credentials credentials);
-
-    public abstract Builder<MethodId> setServiceAddress(String serviceAddress);
-
-    public abstract Builder<MethodId> setPort(int port);
-
-    public abstract Builder<MethodId> setChannel(ManagedChannel channel);
+    public Builder<MethodId> setChannel(ManagedChannel channel) {
+      ChannelProvider provider = new ChannelProvider() {
+        @Override
+        public ManagedChannel getChannel(Executor executor) {
+          return channel;
+        }
+      };
+      return setChannelProvider(provider);
+    }
+
+    public Builder<MethodId> setChannel(ConnectionSettings settings) {
```

Comment:
I have a problem with the arguments not matching the name (setChannel here taking ConnectionSettings)